### PR TITLE
Configurable storage (local vs session)

### PR DESCRIPTION
**Pull request by karelrochelt**

### CODE REVIEW

File: src/connection/session-manager.js [hunk 1]
```diff
@@ -2,13 +2,23 @@ import {
...
+class SessionManager {
+
+  constructor(type) {
+    this.type = type
+  }
 
   _getStorage() {
-    if(window.localStorage) {
-      return window.localStorage
+    // Chrome can return Access Denied error when trying to access window.localStorage
```
Comment (zatziky) [inline src/connection/session-manager.js:12]: What can customers do about it? Isn't in for private mode only?

File: src/connection/session-manager.js [hunk 2]
```diff
@@ -2,13 +2,23 @@ import {
...
+    try {
+      switch(this.type) {
+        case 'local': return window.localStorage
+        case 'session': return window.sessionStorage
+        default: return new TestStorage() // used in tests
+      }
+    } catch(e) {
+      return new TestStorage()
```
Comment (zatziky) [inline src/connection/session-manager.js:20]: If `Access Denied` happens, they get TestStorage?

**Review by zatziky — APPROVED** [review]

File: src/connection/session-manager.js (hunk 1 above)
Comment (karelrochelt) [reply src/connection/session-manager.js:12]: yeah, forgot to mention private mode

File: src/connection/session-manager.js (hunk 2 above)
Comment (karelrochelt) [reply src/connection/session-manager.js:20]: yes, it basically emulates local storage

**Pull request merged**